src/System.IO.Pipelines/src/System/IO/Pipelines/BufferLimitExceededException.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO.Pipelines
+{
+    /// <summary>
+    /// The exception is thrown when buffer limit is exceeded by the writer.
+    /// </summary>
+    public class BufferLimitExceededException : InvalidOperationException
```

Comment:
As new public API, this should go through the normal API review process.

Reply:
cc @terrajobst

Reply:
I edited the original issue too

Reply:
SignalR, for example, wanted to tell people to change signalr related limit